src/spec-node/dockerCompose.ts

```diff
@@ -557,7 +557,7 @@ while sleep 1 & wait $$!; do :; done", "-"${userEntrypoint.map(a => `, ${JSON.st
     init: true` : ''}${user ? `
     user: ${user}` : ''}${Object.keys(env).length ? `
     environment:${Object.keys(env).map(key => `
-      - ${key}=${env[key]}`).join('')}` : ''}${mergedConfig.privileged ? `
+      - "${key}=${env[key].replace(/"/g, '\\"')}"`).join('')}` : ''}${mergedConfig.privileged ? `
```

Comment:
What would be the resulting value? Should we put the `env[key]` in quotes if the quotes the in the values are escaped.

Is it possible to add tests for this scenario?

Reply:
There are additional characters to escape in a double quoted string. I might be easier to use a single quoted string, though I'm not sure how a newline would be represented: https://www.yaml.info/learn/quote.html

Reply:
Hello @chrmarti,

I have made the required change and wrapped the environment properties in single quote instead of double quotes and tested with different combinations of characters in the overriding environment properties in the base container image as below. 

	"containerEnv": {
		"Test_Message": "u?/s\n\ner:p!a,s//sw\\'ord''' ; `K`#a*\nni&sk'a{ S}^e%n(g)u_pta@-",
		"ROSCONSOLE_FORMAT": "[$${severity}] [$${walltime:%Y-%m-%d %H:%M:%S}] [$${node}]: $${message}"
	}
I have few observations:

1. For the newline character \n, its always displayed as blank space and the  first \n is removed in Linux.

Below is how the environment property is printed for the devcontainer image created with docker compose in comparison with the base image used in the docker compose.

`@Kaniska244 ➜ /workspaces/cli (cli_issue_#904_solution) $ docker ps
CONTAINER ID   IMAGE                                                                                     COMMAND                  CREATED             STATUS             PORTS     NAMES
9ee4ae33ecaa   vsc-original-container-c521b00ea40fee92e585b105d9e4f1699ad8216f18220c1b914451f2eafef3b4   "/bin/sh -c 'echo Co…"   About an hour ago   Up About an hour             image-containerenv-issue_devcontainer-devcontainerissues-1
9013d270d6d4   vsc-original-container-c521b00ea40fee92e585b105d9e4f1699ad8216f18220c1b914451f2eafef3b4   "/bin/sh -c 'echo Co…"   About an hour ago   Up About an hour             boring_kapitsa
@Kaniska244 ➜ /workspaces/cli (cli_issue_#904_solution) $ 
@Kaniska244 ➜ /workspaces/cli (cli_issue_#904_solution) $ 
@Kaniska244 ➜ /workspaces/cli (cli_issue_#904_solution) $ 
@Kaniska244 ➜ /workspaces/cli (cli_issue_#904_solution) $ 
@Kaniska244 ➜ /workspaces/cli (cli_issue_#904_solution) $ docker exec 9ee4ae33ecaa bash -c 'echo $Test_Message'
u?/s er:p!a,s//sw\'ord''' ; `K`#a* ni&sk'a{ S}^e%n(g)u_pta@-
@Kaniska244 ➜ /workspaces/cli (cli_issue_#904_solution) $ docker exec 9013d270d6d4 bash -c 'echo $Test_Message'
u?/s er:p!a,s//sw\'ord''' ; `K`#a* ni&sk'a{ S}^e%n(g)u_pta@-
@Kaniska244 ➜ /workspaces/cli (cli_issue_#904_solution) $`


2. In Linux the first '\$' received as a character is omitted. Do we need to change this in the code to preserve it?

Please consider the below example for the same.
`@Kaniska244 ➜ /workspaces/cli (cli_issue_#904_solution) $ docker exec 9ee4ae33ecaa bash -c 'echo $ROSCONSOLE_FORMAT'
[${severity}] [${walltime:%Y-%m-%d %H:%M:%S}] [${node}]: ${message}
@Kaniska244 ➜ /workspaces/cli (cli_issue_#904_solution) $ 
@Kaniska244 ➜ /workspaces/cli (cli_issue_#904_solution) $ docker exec 9013d270d6d4 bash -c 'echo $ROSCONSOLE_FORMAT'
[$${severity}] [$${walltime:%Y-%m-%d %H:%M:%S}] [$${node}]: $${message}`